server/src/main/java/org/elasticsearch/action/bulk/BulkOperation.java

```diff
@@ -282,6 +480,8 @@ public void onNewClusterState(ClusterState state) {
 
             @Override
             public void onClusterServiceClose() {
+                // TODO: There is likely very little we can do about this, and our time in this JVM is likely short,
+                //  should we just get out of here ASAP?
```

Comment:
As mentioned in the comment: There is some weirdness about this edge case that gets introduced when we have two phases of checks.

Originally, if we reached this code, it was before we submitted any operations to the shards, and thus simply failing the whole request is pretty reasonable since it has not even really started yet.

Now, we have a chance of encountering this after we've written our data to the shards, but there were failures that we want to redirect. Simply failing the entire request here feels wrong, since it has mostly succeeded, the only problem is that the failure store feature has not kicked in yet. If the bulk operation had updates mixed into it, the client might see those updates as having not been applied when they have indeed been applied before the cluster service closed.

I'm looking into what is going on when this kind of case happens. My suspicion is that the node is being shut down and we're not long for the world. It would be simple enough to have this call the new `onClusterBlocked` handler, which will be identical logic in the first case above, but in the new case it will simply skip trying to send any failure store redirects and return the results of the bulk (failures included) immediately.